airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -1163,3 +1165,160 @@ def stream_slices(
+
+    name = "GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE_V2"
+    replication_start_date_field = "createdSince"
+    replication_end_date_field = "createdUntil"
+    cursor_field = "createdSince"
+    
+    def __init__(self, url_base: str, aws_signature: AWSSignature, replication_start_date: str, marketplace_id: str, period_in_days: Optional[int], report_options: Optional[str], max_wait_seconds: Optional[int], replication_end_date: Optional[str], authenticator: HttpAuthenticator = None):
+       super().__init__(url_base, aws_signature, replication_start_date, marketplace_id, period_in_days, report_options, max_wait_seconds, replication_end_date, authenticator)
+       self._cursor_value = None
+
+    @property
+    def state(self) -> Mapping[str, Any]:
+        if isinstance(self._cursor_value, str) and self._cursor_value != "" :
+            self._cursor_value = self._cursor_value
+        return {self.cursor_field: self._cursor_value}        
+
+    @state.setter
+    def state(self, value: Mapping[str, Any]):
+        if self.cursor_field not in value:
+            self._cursor_value = None
+        else: 
+            self._cursor_value = value[self.cursor_field]
+
+    def _create_report(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Mapping[str, Any]:
+
+        # For backwards
+        return {"reportId": stream_slice.get("report_id")}
+
+    def stream_slices(
+        self, *, sync_mode: SyncMode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        """
+        From https://developer-docs.amazon.com/sp-api/docs/report-type-values
+        documentation:
+        ```Settlement reports cannot be requested or scheduled.
+            They are automatically scheduled by Amazon.
+            You can search for these reports using the getReports operation.
+        ```
+        """
+        strict_start_date = pendulum.now("utc").subtract(days=90)
+        
+        start_date = max(pendulum.parse(self._replication_start_date), strict_start_date)
+        end_date = pendulum.parse(self._replication_end_date or pendulum.now("utc").date().to_date_string())
+        
+        if sync_mode == SyncMode.incremental:
+            if self._cursor_value:
+                start_date = pendulum.parse(self._cursor_value)
+
+            if stream_state and stream_state.get(self.cursor_field):
+                start_date = pendulum.parse(stream_state[self.cursor_field])
+
+            start_date = max(start_date, strict_start_date)
+        
+        if end_date < strict_start_date:
+            end_date = strict_start_date
+
+        if end_date < start_date:
+            return []
+
+        params = {
+            "reportTypes": self.name,
+            "pageSize": 100,
```

Comment:
```suggestion
            "pageSize": self.page_size,
```
and create the variable